framework/src/plugins/base_plugin.ts

```diff
@@ -32,12 +34,86 @@ export interface InstantiablePlugin<T, U = object> {
 	new (...args: U[]): T;
 }
 
+interface BaseTransactionJSON {
+	readonly type: number;
+	readonly nonce: string;
+	readonly fee: string;
+	readonly senderPublicKey: string;
+	readonly signatures: Array<Readonly<string>>;
+
+	readonly asset: string;
+}
+
+interface TransactionJSON {
+	readonly type: number;
+	readonly nonce: string;
+	readonly fee: string;
+	readonly senderPublicKey: string;
+	readonly signatures: Array<Readonly<string>>;
+
+	readonly id: string;
+	readonly asset: object;
+}
+
+const decodeTransactionToJSON = (
+	transactionBuffer: Buffer,
+	baseSchema: Schema,
+	assetsSchemas: { [key: number]: Schema },
+): TransactionJSON => {
+	const baseTransaction = codec.toJSON<BaseTransactionJSON>(
```

Comment:
`toJSON` is not the correct method to use. it should be `decodeJSON`.
toJSON is for object = JSON conversion.
Also, do we really want to convert to JSON here? if there are specific calculation done, for example `fee`, it needs to converted back to BigInt again.